pkg/health/checker/checker.go

```diff
@@ -64,28 +62,49 @@ func NewConsulHealthChecker(client consulutil.ConsulClient) ConsulHealthChecker
 	}
 }
 
-func (c consulHealthChecker) WatchNodeService(
+func (c consulHealthChecker) WatchPodOnNode(
 	nodename types.NodeName,
-	serviceID string,
-	resultCh chan<- health.Result,
-	errCh chan<- error,
+	podID types.PodID,
 	quitCh <-chan struct{},
-) {
-	defer close(resultCh)
+) (chan health.Result, chan error) {
+	resultCh := make(chan health.Result)
+	errCh := make(chan error)
 
-	for {
-		select {
-		case <-quitCh:
-			return
-		case <-time.After(1 * time.Second):
-			kvCheck, err := c.consulStore.GetHealth(serviceID, nodename)
-			if err != nil {
-				errCh <- err
-			} else {
-				resultCh <- consulWatchToResult(kvCheck)
+	key := consul.HealthPath(podID.String(), nodename)
+
+	wsOut := make(chan *api.KVPair) // closed by WatchSingle
+	wsQuit := make(chan struct{})
+	defer close(wsQuit)
```

Comment:
you probably want this inside the goroutine, otherwise it's going to be closed immediately!

Reply:
🙃